File: e-commerce-backend/src/main/java/com/TechKun/model/Promotion.java
package com.TechKun.model;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDate;
import java.util.List;

import com.TechKun.model.enums.PromotionType;

@Entity
@Data
@Table(name = "Promotion")
public class Promotion {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "promotion_id", nullable = false)
    private Integer promotionId;

    @Column(name = "description", nullable = false)
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(name = "promotion_type", nullable = false)
    private PromotionType promotionType;

    @Column(name = "discount_value", nullable = false)
    private Double discountValue;

    @Column(name = "valid_from")
    private LocalDate validFrom;

    @Column(name = "valid_till")
    private LocalDate validTill;

    @Column(name = "minimum_order_value")
    private Double minimumOrderValue;

    @Column(name = "max_uses")
    private Integer maxUses;

    @Column(name = "usage_per_customer")
    private Integer usagePerCustomer;

    @ManyToMany
    @JoinTable(name = "promotion_category",
        joinColumns = @JoinColumn(name = "promotion_id"),
        inverseJoinColumns = @JoinColumn(name = "category_id")
    )
    private List<Category> categories;
}

File: e-commerce-frontend/src/types/domains/promotion.ts
import { Category } from "./category";

export enum PromotionType {
    PERCENTAGE = "PERCENTAGE",
    FLAT = "FLAT"
}

export interface Promotion {
    promotionId: number;
    description: string;
    promotionType: PromotionType;
    discountValue: number;
    validFrom: Date;
    validTill: Date;
    minimumOrderValue: number;
    maxUses: number;
    usagePerCustomer: number;
    categories: Category[];
}

export interface PromotionDetails {
    promotionId: number;
    description: string;
    promotionType: PromotionType;
    imageUrl?: string;
    discountValue: number;
    validFrom?: Date;
    validTill?: Date;
    minimumOrderValue?: number;
    maxUses?: number;
    usagePerCustomer?: number;
    categories: {
        categoryId: number;
        name: string;
    }[];
}

export interface PromotionDTO {
    description: string;
    promotionType: PromotionType;
    discountValue: number;
    validFrom?: Date;
    validTill?: Date;
    minimumOrderValue?: number;
    maxUses?: number;
    usagePerCustomer?: number;
    categoryIds: number[];
}